mmdet/apis/train.py

```diff
@@ -68,6 +68,54 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def autoscale_lr(cfg, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        logger (logging.Logger): logger.
+    """
+    auto_scale_lr_flag = cfg.get('auto_scale_lr', False)
+    if auto_scale_lr_flag is False:
+        logger.info('Automatic scaling of learning rate (LR)'
+                    ' has been disabled.')
+        return
+
+    # default batch size of pre-train model
+    default_batch_size = cfg.get('default_batch_size', None)
+    if default_batch_size is None:
+        logger.warning('Can not find default_batch_size in config file, '
+                       'please check if it existed. '
+                       'Disable automatic scaling of learning rate.')
+        return
+
+    # calculate the batch size
+    num_gpus = len(cfg.gpu_ids)
```

Comment:
should use world_size

Reply:
we should do autoscale_lr after distributed env is initialized.

Reply:
In file `tools\train.py`, when distributed enable, the `cfg.gpu_ids` will get info from `dist.get_world_size()`. And those steps process before function `scale_lr`. So it's safe to get the gpu number using `len(cfg.gpu_ids)` here.

![image](https://user-images.githubusercontent.com/25873202/158172899-34ae87cd-eb92-4953-842b-aebdb4410778.png)

![image](https://user-images.githubusercontent.com/25873202/158173097-2b44b71c-e23a-4313-90de-d02c8d0d3510.png)

Reply:
Hi @PeterH0323. After discussion, we consider using `word_size` is safer than directly using `args.gpu_ids`.

Reply:
Done